### PR TITLE
Updating Readme

### PR DESCRIPTION
**Pull request by kartsree**
### All Submissions:

* [ ] Have you followed the guidelines in our Contributing document?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [ ] Does your submission pass tests?
2. [ ] Have you lint your code locally prior to submission?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 [![CircleCI](https://circleci.com/gh/shipt/specter/tree/master.svg?style=svg&circle-token=00862a45c139d2a1a28c93ca104e38a7a6e3e83e)](https://circleci.com/gh/shipt/specter/tree/master)
 ![Specter Screenshot](/readmeFiles/SpecterScreenShot.gif)
 
-Specter is an attack map style visualization that parses NGINX access logs. Specter then displays the source ip's location, the NGINX's ip's location, and the http status on a world map. 
+Specter is an attack map style visualization that parses NGINX access logs. Specter then displays the source ip's location, the NGINX's ip's location, and the http status on a world map. Additionally allows filtering the traffic based on services.
```
Comment (wesleyorama-shipt) [inline README.md:6]: We should update the wiki with steps required to use filtering and a quick run down of how it works.

File: internal/dataServer/dataServer.go [hunk 2]
```diff
@@ -121,7 +122,8 @@ func processLog(reader ngninxLogReader, ip net.IP) (msg, error) {
 		}).Warn("error getting http_x_forwarded_for from the access.log")
 		return msg{}, nil
 	}
-	return msg{SrcIP: ra, DstIP: ip.String(), HTTPStatus: s, XFwdFor: x}, nil
+	p, err := rec.Field("rand_host")
+	return msg{SrcIP: ra, DstIP: ip.String(), HTTPStatus: s, XFwdFor: x, HostProxy: p}, nil
```
Comment (wesleyorama-shipt) [inline internal/dataServer/dataServer.go:126]: make sure to update dataServer_test.go to include tests for having and not having p be zero value

File: README.md (hunk 1 above)
Comment (kartsree) [reply README.md:6]: @wesleyorama-shipt Updated the Wiki on the new dropdown feature and how to leverage it when running locally

File: internal/dataServer/dataServer.go (hunk 2 above)
Comment (kartsree) [reply internal/dataServer/dataServer.go:126]: Added test cases when we have a host_proxy and when the host_proxy filed is empty

File: web/public/app.dev.js [hunk 3]
```diff
@@ -64,21 +67,34 @@ class App extends React.PureComponent {
...
         if (evt.data === '') return
         const wsData = JSON.parse(evt.data)
+
+        if (host_proxy_list.includes(wsData['HostProxy']) === false) {
+          host_proxy_list.push(wsData['HostProxy'])
+        }
+
+        var result = host_proxy_list.map(e => ({id: e, name: e}))
+
+        this.setState({host_list: result});
```
Comment (jflayhart) [inline web/public/app.dev.js:92]: this may be ok... but this is going to fire ALOT, so if you run into performance issues (react working too hard) then you may need to throttle this method to only happen every 200ms or something like that.

File: web/public/app.dev.js [hunk 4]
```diff
@@ -64,21 +67,34 @@ class App extends React.PureComponent {
...
+
+        if (wsData['HostProxy'])  {
+          if (this.state.dropdown != "all" && wsData['HostProxy'] != this.state.dropdown) return
+        }
```
Comment (jflayhart) [inline web/public/app.dev.js:96]: personally, i'd leave comments to help others understand the conclusion that lead us to this logic.
